Modules/Managers/SpawnManager.cs

```diff
@@ -0,0 +1,116 @@
+
+using CounterStrikeSharp.API.Core;
+using CounterStrikeSharp.API.Modules.Utils;
+using RetakesPlugin.Modules.Configs;
+using RetakesPlugin.Modules.Enums;
+
+namespace RetakesPlugin.Modules.Managers;
+
+public class SpawnManager
+{
+    private readonly Translator _translator;
+    private readonly MapConfig _mapConfig;
+	private readonly Dictionary<Bombsite, Dictionary<CsTeam, List<Spawn>>> _spawns = new();
+	private readonly bool _spawnsInitialized = false;
+
+    public SpawnManager(Translator translator, MapConfig mapConfig)
+    {
+        _translator = translator;
+		_mapConfig = mapConfig;
+
+		foreach (var spawn in _mapConfig.GetSpawnsClone())
+		{
+			if (!_spawns.ContainsKey(spawn.Bombsite))
+			{
+				_spawns.Add(spawn.Bombsite, new Dictionary<CsTeam, List<Spawn>>());
+			}
+
+			if (!_spawns[spawn.Bombsite].ContainsKey(spawn.Team))
+			{
+				_spawns[spawn.Bombsite].Add(spawn.Team, new List<Spawn>());
+			}
+
+			_spawns[spawn.Bombsite][spawn.Team].Add(spawn);
+		}
+		_spawnsInitialized = true;
+    }
+
+	public List<Spawn> GetSpawns(Bombsite bombsite, CsTeam? team = null)
+	{
+		if (!_spawnsInitialized)
```

Comment:
Checking `if (!_spawnsInitialized)` is redundant. 

You **might** need to check to see if `_spawns[bombsite]` contains the team value you're checking for.